Monika After Story/game/script-topics.rpy

```diff
@@ -15990,3 +15990,106 @@ label monika_stargazing:
     show monika 5kkbsa
     pause 2.0
     return
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="monika_player_eyesight",
+            category=["you"],
+            prompt="Wearing glasses",
+            random=True
+        )
+    )
+
+default persistent._mas_pm_uses_eyesight_aids = None
+
+label monika_player_eyesight:
+    m 3lsb "Hey, [player], I was wondering about something lately."
+    m 1esa "Do you wear glasses, or maybe contact lenses?{nw}"
+    $ _history_list.pop()
+    menu:
+        m "Do you wear glasses, or maybe contact lenses?{fast}"
+
+        "I always wear glasses.":
+            call monika_player_eyesight_awea(glasses=True)
+
+        "I wear contact lenses.":
+            call monika_player_eyesight_awea(glasses=False)
+
+        "I wear cosmetic glasses.":
+            call monika_player_eyesight_cg
+
+        "I wear them when needed.":
+            call monika_player_eyesight_eawn
+
+        "Nope, I don't.":
+            call monika_player_eyesight_n
+
+    return "derandom"
+
+# Always Wears Eyesight Aids
+label monika_player_eyesight_awea(glasses=False):
+    $ persistent._mas_pm_uses_eyesight_aids = True
+    m 2ekc "Oh, I see. You've got poor eyesight, haven't you?"
+    m 2eka "I really hope you see better with them on."
+    if glasses:
+        m 3eub "And there's actually a bright side of wearing glasses - I'm sure you look so smart and intelligent, ehehe~"
+    m 1eka "Please take good care of yourself, [player], even when it comes to something minor like this."
+    m 1esa "If you've been on your computer for a while, maybe you could consider taking a little break to let your eyes rest."
+    m 3hsa "It'll take you just a few minutes a day, but I believe it would really help."
+    m 1ekc "Although... if it gets really bad, go see a doctor, okay?"
+    m 1eka "I really want my sweetheart to see this beautiful world just as clear as I do~"
+    return
+
+# Cosmetic Glasses
+label monika_player_eyesight_cg:
+    $ persistent._mas_pm_uses_eyesight_aids = False
+    m 5ekbsa "Aw, [player], I wish I really could see your face with them on."
+    m 5hubsa "I bet you look so smart, intelligent and confident~"
+    m 1sua "Maybe I could even get myself a pair and try them out, too, ehehe."
+    m 1tuu "What do you think, [mas_get_player_nickname()]? Would they suit me well, hmm?"
+    return
+
+# Eyesight Aids When Needed
+label monika_player_eyesight_eawn:
+    m 3rka "I see... But [player], is it, by any chance, because of your eyesight?{nw}"
+    $ _history_list.pop()
+    menu:
+        m "I see... But [player], is it, by any chance, because of your eyesight?{fast}"
+
+        "Yes...":
```

Comment:
If you tell Monika you wear glasses when you need them, I think it's pretty safe to assume that you wear them because you _need_ them to see.

Reply:
I'm sorry, but I was previously told the opposite in one of the discussions about this exact topic about a month back.

Reply:
That was me.
The point is that some people like me, can wear glasses for a non-essential purpose (in my case, resting my eyes when looking at a screen).

I can see just fine without them, and I can still spend entire days in front of a computer without wearing them, but they're still a nice thing to have (saves me some headaches).

You could perhaps make it more clear by changing `"I wear them when needed."` by something like `"I wear them occasionally."` or `"I wear them depending on the situation."`.

Reply:
That is not a problem at all. 

What about contact lenses, though? Do they fall under this option? Can they be worn occasionally?

If not, then I will use "I wear *glasses* occasionally".